File: java/src/main/java/toolgood/algorithm/internals/functions/csharp/Function_GUID.java
package toolgood.algorithm.internals.functions.csharp;

import java.util.UUID;
import java.util.function.BiFunction;

import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_GUID extends Function_0 {
    public Function_GUID() {
    }

    @Override
    public String Name() {
        return "Guid";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        return Operand.Create(UUID.randomUUID().toString());
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append("GUID()");
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.TEXT;
    }
}

File: csharp/ToolGood.Algorithm/Internals/Functions/Csharp/Function_GUID.cs
using System;
using System.Text;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.Csharp
{

	internal sealed class Function_GUID : FunctionBase
	{
		public Function_GUID()
		{
		}

		public override string Name => "Guid";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			return Operand.Create(System.Guid.NewGuid().ToString());
		}
		public override void ToString(StringBuilder stringBuilder, bool addBrackets)
		{
			stringBuilder.Append("GUID()");
		}
		public override OperandType GetResultType()
		{
			return OperandType.TEXT;
		}
	}


}
